workflow/controller/controller.go

```diff
@@ -306,10 +306,6 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, workflowTTLWo
 
 	// Create Synchronization Manager
 	wfc.createSynchronizationManager(ctx)
-	// init managers: throttler and SynchronizationManager
-	if err := wfc.initManagers(ctx); err != nil {
-		log.Fatal(err)
-	}
```

Comment:
I'm not sure that moving this is correct -- the Workflows that the Informers capture may skip any throttling or synchronization

We could potentially skip calling this entirely if the Informer will add them to the managers in the same order though 🤔

Reply:
yes, moving wil skip throttling and synchronization. I have added a new InitEventHandler to handle throttling and synchronzation first, then do `addWorkflowInformerHandlers`, PTAL, thanks. @agilgur5

Reply:
thanks for your reminder, I have updated and the order of `addWorkflowInformerHandlers` has been readjusted to ensure that `initManagers` is completed before informer event handling.